### PR TITLE
Full screen container opening (Checkpoint) (BLOCKED) [GOL-33]

### PR DESCRIPTION
**Pull request by jboolean**
This PR adds support for viewing sub-graphs (containers). They can be opened by double-clicking a node with sharp corners and navigated using breadcrumbs at the top. Styled using styles and ideas proposed in my photoshop comp.

**Note:** This PR is **not to be merged**. It is branched off of **#30 zoom** as I needed GraphCanvas from there, and thus is **blocked** by the approval of #30. A new PR will be created once those changes are in and discussion here is finished.

There are a few things I don't like in this PR, and I'd like some feedback:
- Passing the opening function as a prop. I tried to think about how to make some kind of global event bus, or implementation of the Command Pattern that would help decouple future event generators from receivers. Future events may include selection of an object, which would interest editor panels; grouping, which would interest the data models and may be generated via a button or a keyboard shortcut; or other such events with multiple generators or receivers. 
- The parallel array of container vertices in graphCanvas. Yuck. Better idea?
- Making my own doubleclick. 
- Saving pan and zoom like that seems inelegant, but I don't know a better way. I mean, you could leave the graphs in the DOM, hidden, but then it would be rendering potentially large graphs you can't see.
- I'd like to add an animation where the vertex zooms in to indicate traversing into it. I tried this with CSS animations, but could only get it to zoom to the corner; I couldn't figure out how to animate a scale about the center.

### CODE REVIEW

**Comment by jboolean** [comment]
Silence is consent.

File: lib/utilities/mouseDownDrag.js [hunk 1]
```diff
@@ -93,6 +93,9 @@ var onMouseMove = function(name, pseudoDragCallback, scaleFactor, event) {
  * @returns {void}
  */
 var onMouseUp = function(name, pseudoDragEndCallback, event) {
+  if (!this.isMounted()) {
+    return;
+  }
```
Comment (RobertWarrenGilmore) [inline lib/utilities/mouseDownDrag.js]: Are things unmounting during drags?

File: lib/utilities/pseudoDoubleClick.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    cb(e);
+    newState[stateKey] = false;
+  } else {
+    newState[stateKey] = Date.now();
+  }
+
+  this.setState(newState);
+};
+
+module.exports = doubleClick;
```
Comment (RobertWarrenGilmore) [inline lib/utilities/pseudoDoubleClick.js]: Wow. I'm surprised that we need this. SVG just doesn't support much fancy mousery, does it?

File: lib/viewcomponents/graph.jsx [hunk 3]
```diff
@@ -1,4 +1,5 @@
 var React = require('react/addons');
+var Lang = require('../utilities/lang');
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/graph.jsx]: This is never used.

File: lib/utilities/mouseDownDrag.js (hunk 1 above)
Comment (jboolean) [inline lib/utilities/mouseDownDrag.js]: A "drag" is anything between a mouse down and a mouse up. Between that you could have opened a new container, unmounting the previous Graph.

File: lib/utilities/pseudoDoubleClick.js (hunk 2 above)
Comment (jboolean) [inline lib/utilities/pseudoDoubleClick.js]: Well, more of a React problem and a Javascript problem in general. Can't bind click and dblclick both.

File: lib/viewcomponents/graph.jsx (hunk 3 above)
Comment (jboolean) [inline lib/viewcomponents/graph.jsx]: Thanks. Removing.

**Pull request closed without merging**